forui/lib/src/widgets/tile/tile.dart

```diff
@@ -0,0 +1,323 @@
+import 'package:flutter/foundation.dart';
+import 'package:flutter/widgets.dart';
+import 'package:forui/forui.dart';
+import 'package:forui/src/foundation/tappable.dart';
+import 'package:forui/src/widgets/tile/tile_content.dart';
+import 'package:meta/meta.dart';
+
+/// A tile that is typically used to group related information together.
+///
+/// Multiple tiles can be grouped together in a [FTileGroup]. Tiles grouped together will be separated by a divider,
+/// specified by a [FTileDivider].
+///
+/// See:
+/// * https://forui.dev/docs/tile/tile for working examples.
+/// * [FTileGroup] for grouping tiles together.
+/// * [FTileStyle] for customizing a tile's appearance.
+class FTile extends StatelessWidget {
+  /// The tile's style. Defaults to the ancestor tile group's style if present, and [FThemeData.tileGroupStyle] otherwise.
+  ///
+  /// Provide a style to prevent inheriting from the ancestor tile group's style.
+  final FTileStyle? style;
+
+  /// Whether the FTile is enabled. Defaults to true.
+  final bool enabled;
+
+  /// The semantic label.
+  final String? semanticLabel;
+
+  /// A callback for when the tile is pressed.
+  ///
+  /// The tile is not hoverable if both [onPress] and [onLongPress] are null.
+  final VoidCallback? onPress;
+
+  /// A callback for when the tile is long pressed.
+  ///
+  /// The tile is not hoverable if both [onPress] and [onLongPress] are null.
+  final VoidCallback? onLongPress;
+
+  /// The child.
+  final Widget child;
+
+  /// Creates a [FTile].
+  ///
+  /// ```
+  /// -----------------------------------------------------
+  /// | [prefixIcon] [title]       [details] [suffixIcon] |
+  /// |              [subtitle]                           |
+  /// ----------------------------------------------------
+  /// ```
+  ///
+  /// ## Overflow behavior
+  /// If the tile's content overflows and `details` is text, it'll be truncated first. Otherwise, `title` and `subtitle`
+  /// will be truncated first.
+  FTile({
+    required Widget title,
+    this.style,
+    this.enabled = true,
+    this.semanticLabel,
+    this.onPress,
+    this.onLongPress,
+    Widget? prefixIcon,
+    Widget? subtitle,
+    Widget? details,
+    Widget? suffixIcon,
```

Comment:
I don't have strong opinions on this, but is it better to make the `suffixIcon` default to `FIcon(FAssets.icons.chevronRight)`? Just thought it might be worth bring up.

Reply:
I have mixed feeling about making FIcon(FAssets.icons.chevronRight) the default, I was thinking of potentially adding another name constructor, maybe something like `FTile.navigation(...)` that does that.